Integración de últimos cambios hechos al Front-end

Kasuk1 opened the pull request:
Hola Mateo, el siguiente PR es un cambio relevante en temas de diseño de la app. Somos conscientes de algunas importaciones de ant-design que hemos dejado flotando, sin embargo por ahora vemos conveniente dejarlas allí por el momento.

Kasuk13 committed:
Estructura de directorios

Kasuk13 committed:
Editando Login Formulario

Kasuk13 committed:
Estructura de carpetas y archivos modificado

Kasuk13 committed:
Routes y estructura básica

Kasuk13 committed:
Routes y estructura básica

JeanMarco173 committed:
Home

Kasuk13 committed:
Login, Registro y Mascotas(List y Item) Avance-Igor

Kasuk13 committed:
Login, Registro y Mascotas(List y Item) Avance-Igor

Kasuk13 committed:
MascotasDetaille, listaMascota, data/mascotas Agregado y routes/AppRoute Modificado

Kasuk13 committed:
MascotasDetaille, listaMascota, data/mascotas Agregado y routes/AppRoute Modificado

Kasuk13 committed:
Cambios modificados MascotaDetalle

Kasuk13 committed:
Cambios modificados MascotaDetalle

Kasuk13 committed:
Cambios modificados lineas en blanco

Kasuk13 committed:
Cambios modificados lineas en blanco

Kasuk13 committed:
Cambios modificados lineas en blanco AppRoute

Kasuk13 committed:
Cambios modificados lineas en blanco AppRoute

rivaldo-ortega committed:
Merge pull request #2 from marceg88/branch-igor

Branch igor

rivaldo-ortega committed:
registro de mascotas y lista de veterinarias

JeanMarco173 committed:
Agrega Home Principal

rivaldo-ortega committed:
veterinaria

rivaldo-ortega committed:
correciones rivaldo

JeanMarco173 committed:
Agrega Fixes

JeanMarco173 committed:
Merge branch 'develop' into develop-jm

JeanMarco173 committed:
Merge pull request #11 from marceg88/develop-jm

Add Home

marceg88 committed:
marcela

Kasuk13 committed:
Ultimos cambios develop

Kasuk13 committed:
Front-end rematered, Redux Toolkit added

Kasuk13 committed:
Deleted not used warnings

Kasuk13 committed:
_redirects file added to public for client-side routing

mhbuiles commented on src/components/globales/footer/Footer.css:
```diff
@@ -0,0 +1,51 @@
+        flex-direction: row;
+        gap: 1.5rem;
+    }
+}
```
Se recomienda dejar una línea en blanco al final de cada archivo

mhbuiles commented on src/components/globales/navbar/Navbar.css:
```diff
@@ -0,0 +1,122 @@
+        height: 7rem;
+    }
+
+
```
No es necesario dejar tantas líneas en blanco

mhbuiles commented on src/components/globales/navbar/Navbar.js:
```diff
@@ -0,0 +1,51 @@
+                        <li className={className}><a href='https://www.google.com/'>
+                            <i className="fab fa-instagram"></i>
+                        </a></li>
+
```
Esta línea en blanco no la necesitamos

mhbuiles commented on src/components/globales/navbar/Navbar.js:
```diff
@@ -0,0 +1,51 @@
+        </>
+
+    );
+};
```
Se recomienda dejar una línea en blanco al final de cada archivo

mhbuiles commented on src/components/home/general/HomeGeneral.css:
```diff
@@ -0,0 +1,290 @@
+    }
+}
+
+
```
No dejemos tantas líneas en blanco entre bloques de código

mhbuiles commented on src/components/home/general/HomeGeneral.js:
```diff
@@ -0,0 +1,114 @@
+                            src={touch ? welcomeDogImage : welcomeCatImage}
+                            onMouseEnter={() => setTouch(true)}
+                            onMouseLeave={() => setTouch(false)}
+                            alt=''
```
Es bueno que agreguemos un string a este atributo, esto ayudará a mejorar la accesibilidad de la aplicación para personas que utilicen lectores de pantalla.

mhbuiles commented on src/components/home/usuario/HomeUsuario.js:
```diff
@@ -0,0 +1,7 @@
+export const HomeUsuario = () => {
```
Por qué tenemos este componente vacío?

mhbuiles commented on src/components/login/formularios/LoginFormulario.css:
```diff
@@ -0,0 +1,30 @@
+    max-width: 20rem;
+}
+/* 
+@media only screen and (min-width: 37.5em)  */
```
Por que no borramos este comentario?

mhbuiles commented on src/components/login/formularios/LoginFormulario.js:
```diff
@@ -0,0 +1,85 @@
+    const dispatch = useDispatch();
+
+    const onFinish = (values) => {
+        console.log('Received values of form: ', values);
```
No dejemos console.log en el código

mhbuiles commented on src/components/mascotas/detalleMascota/MascotaDetalle.js:
```diff
@@ -0,0 +1,64 @@
+
+    const { petId } = useParams();
+
+    const { name, description, image, historial } = mascotas.find(mascota => mascota.id === Number(petId));
```
Esto es provisional mientras traemos esta data del servidor??

Kasuk1 replied:
Hola Mateo, eso es provisional por el momento hasta q utilizemos la API.

mhbuiles commented on src/components/mascotas/detalleMascota/MascotaDetalleModal.css:
```diff
@@ -0,0 +1,11 @@
+.modal--info {
```
Necesitamos estas clases vacías? A esto hacían referencia con su comentario inicial?

Kasuk1 replied:
Olvidé borrarlas. Respecto al comentario inicial resulta invalidado debido a que el despliegue a Netlify me obligó a quedarme sin warnings.

mhbuiles commented on src/components/mascotas/detalleMascota/MascotaDetalleModal.js:
```diff
@@ -0,0 +1,38 @@
+                    </ul>
+                </div>
+            ),
+            onOk() { },
```
No debemos pasarla algo a onOk?

Kasuk1 replied:
Realmente solo es para ver el detalle del servicio de la mascota. Sin embargo acabo de revisar y no es un método necesario si no se realizará alguna acción al dar ok. Ya lo borré. Gracias.

mhbuiles commented on src/components/mascotas/listaMascotas/MascotaLista.css:
```diff
@@ -0,0 +1,41 @@
+.section__pets--description {
+    max-width: 50rem;
+}
+.section__pets--title {
```
Aquí veo más clases vacías

mhbuiles commented on src/components/mascotas/listaMascotas/MascotaLista.js:
```diff
@@ -0,0 +1,37 @@
+                    </button>
+                </div>
+                <div className='section__pets--list'>
+                    {mascotas.map(mascota => <Mascota key={mascota.id} {...mascota} />)}
```
Esto es provisional mientras conectamos con el servidor??

Kasuk1 replied:
Eso es provisional.

mhbuiles commented on src/components/mascotas/mascota/Mascota.css:
```diff
@@ -0,0 +1,29 @@
+    padding: 1.5rem 2rem 4rem 2rem;
+}
+.pet-card__name {
+    
```
Clases vacías

mhbuiles commented on src/components/mascotas/mascota/Mascota.js:
```diff
@@ -0,0 +1,25 @@
+                <p className='pet-card__description paragraph'>{description}</p>
+            </div>
+        </div>
+
```
No dejemos tantas líneas en blanco innecesarias

mhbuiles commented on src/components/mascotas/registroMascotas/MascotaRegistro.js:
```diff
@@ -0,0 +1,98 @@
+export const MascotaRegistro = () => {
+
+  const onFinish = (values) => {
+    console.log(values);
```
No dejemos console.log, ya a estas alturas del bootcamp tenemos que ir completando los flujos, qué se va a hacer al enviar el formulario?

Kasuk1 replied:
Es provisional. Justo ya empezamos a integrarlo

mhbuiles commented on src/components/mascotas/registroMascotas/MascotaRegistro.js:
```diff
@@ -0,0 +1,98 @@
+    console.log(values);
+  };
+  const onFinishFailed = (errorInfo) => {
+    console.log('Failed:', errorInfo);
```
Ya estas funciones deben hacer lo que les corresponde, no dejemos console.log

Kasuk1 replied:
Se procederá a borrar los console.log cuando se integre a la API

mhbuiles commented on src/components/mascotas/registroMascotas/MascotaRegistro.js:
```diff
@@ -0,0 +1,98 @@
+  };
+
+  const normFile = (e) => {
+    console.log('Upload event:', e);
```
No console.log

mhbuiles commented on src/components/mascotas/registroMascotas/MascotaRegistro.js:
```diff
@@ -0,0 +1,98 @@
+
+  return (
+    <div className='pet-register'>
+
```
Estas líneas en blanco no son necesarias

mhbuiles commented on src/components/registro/formularios/RegistroFormulario.js:
```diff
@@ -0,0 +1,164 @@
+
+    /* FORM VALIDATIONS */
+    const onFinish = (values) => {
+        console.log('Success:', values);
```
No dejemos console.log, ya a estas alturas del bootcamp tenemos que ir completando los flujos, qué se va a hacer al enviar el formulario?

mhbuiles commented on src/components/registro/formularios/RegistroFormulario.js:
```diff
@@ -0,0 +1,164 @@
+    };
+
+    const onFinishFailed = (errorInfo) => {
+        console.log('Failed:', errorInfo);
```
No dejemos console.log, ya a estas alturas del bootcamp tenemos que ir completando los flujos, qué se va a hacer al enviar el formulario? Ya tenemos que conectarnos con nuestro servidor

mhbuiles commented on src/components/registro/formularios/RegistroFormulario.js:
```diff
@@ -0,0 +1,164 @@
+    /* FORM VALIDATIONS */
+
+    const normFile = (e) => {
+        console.log('Upload event:', e);
```
No console.log

mhbuiles commented on src/components/servicios/detalleServicio/ServicioDetalle.js:
```diff
@@ -0,0 +1,7 @@
+export const ServicioDetalle = () => {
```
Por qué creamos componentes vacíos? Mejor los agregamos a los commit cuando ya hagan algo

mhbuiles commented on src/components/servicios/listaServicios/ServiciosLista.js:
```diff
@@ -0,0 +1,37 @@
+                    </button>
+                </div>
+                <div className='section__services--list'>
+                    {servicios.map(servicio => <Servicio key={servicio.id} {...servicio} />)}
```
Ya debemos ir conectando al servidor.

mhbuiles commented on src/components/servicios/solicitudServicio/ServicioSolicitud.js:
```diff
@@ -0,0 +1,104 @@
+export const ServicioSolicitud = () => {
+
+  const formSuccess = (datos) => {
+    console.log("formulario enviado", datos);
```
Estos endpoints ya están listos en el backend? Si es así, debemos ir conectando

mhbuiles commented on src/components/servicios/solicitudServicio/ServicioSolicitud.js:
```diff
@@ -0,0 +1,104 @@
+    console.log("formulario enviado", datos);
+  }
+  const formFailed = (error) => {
+    console.log("error", error);
```
No console.log

mhbuiles commented on src/components/servicios/solicitudServicio/ServicioSolicitud.js:
```diff
@@ -0,0 +1,104 @@
+      </Form>
+    </div>
+
+
```
No dejemos esta cantidad de líneas en blanco

mhbuiles commented on src/components/veterinarias/listaVeterinarias/VeterinariasLista.js:
```diff
@@ -0,0 +1,30 @@
+        </div>
+
+        <div className='section__veterinaries--list'>
+          {veterinarias.map((veterinaria) => <Veterinaria key={veterinaria.id} {...veterinaria} />)}
```
Esta data va a venir de la base de datos??

Kasuk1 replied:
No está planeado debido a que ya tenemos un CRUD. Sin embargo, está por decidirse sea o no en duro.

mhbuiles commented on src/routes/AppRoutes.js:
```diff
@@ -0,0 +1,45 @@
+
+    return (
+        <BrowserRouter>
+
```
No dejemos estas líneas en blanco, se ve un poco desorganizado

mhbuiles commented on src/routes/AppRoutes.js:
```diff
@@ -0,0 +1,45 @@
+                <Routes>
+
+                    <Route path='/login' element={
+                        <PublicRoutes>
```
Habría forma de dejar todas las rutas públicas anidadas dentro de una sola etiqueta PublicRoutes?

Kasuk1 replied:
Si se puede, sin embargo nos evitamos importar componentes a public o private, y solo dejamos la lógica definida para ambos, desestructurando sus children.

mhbuiles commented on src/services/MyLittleFriendAPI/MyLittleFriendAPI.js:
```diff
@@ -0,0 +1,20 @@
+        });
+        const json = await response.json();
+        token = json.token;
+        console.log(token)
```
No console.log

mhbuiles commented on src/services/MyLittleFriendAPI/MyLittleFriendAPI.js:
```diff
@@ -0,0 +1,20 @@
+
+const MLFURI = 'https://mylittlefriend.herokuapp.com';
```
Esto debe ir en las variables de entorno y ese archivo no se debe subir al repo, no debe estar ese tipo de información en el código

mhbuiles commented on src/store/uiSlice/ui.slice.js:
```diff
@@ -0,0 +1,19 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+
```
No debe haber un initialstate?

Kasuk1 replied:
Aún esta por verse, ya que no hemos agregado estados que cambiaran en la ui. Sin embargo, de no necesitarlo se eliminará el slice.

mhbuiles commented on src/store/userSlice/user.slice.js:
```diff
@@ -0,0 +1,50 @@
+    name: 'user',
+    initialState: {
+        user: {
+
```
Por qué user es vacío? Qué pasa si iniciamos con null?

Kasuk1 replied:
Por ahora lo hemos dejado vacío porque necesitaremos del consumo de la API primero. Justo ahora trabajando en eso.

mhbuiles commented on src/store/userSlice/user.slice.js:
```diff
@@ -0,0 +1,50 @@
+                .addCase(signIn.fulfilled, (state, action) => {
+                    state.signInState.loading = false;
+                    state.signInState.error = false;
+                    console.log(action.payload)
```
No console.log

Kasuk1 replied:
Solo fue para asegurarse del response del método asincrono.

Kasuk1 replied:
Por ahora dejaremos console.log en los estados de los extraReducers. Al final ya quitaremos todos los q implementemos correctamente.

mhbuiles reviewed (changes requested):
No deberíamos subir imágenes a los repositorios

Kasuk13 committed:
Login, MascotasUsuario, MscotaDetalle unidas y renderizadas con API

Kasuk13 committed:
PetListLoading, PetCardLoading,
PetDetailLoading agregados y sincronizados en estado pendiente(cargando) con thunk

Kasuk13 committed:
Imagenes de assets eliminadas

Kasuk13 committed:
PetDetailLoading cambio color

Kasuk13 committed:
Veterinarias y Servicios componentes y servicios actualizados y creados

Kasuk13 committed:
Warnings eliminados para despliegue

Kasuk13 committed:
Cambio pets.map MascotaLista.js

Kasuk13 committed:
Cambios revertidos pets

Kasuk13 committed:
Agregado arreglo de manejo de errores petición de obtener pets by owner

Kasuk13 committed:
Agregado subida de imágenes para usuario y mascota

Kasuk13 committed:
Payment/Card Register agregados con servicios y gestión en redux

Kasuk13 committed:
Agregado Payment con RequestService, Pet Update y Delete

Kasuk13 committed:
Historial médico en detalle de mascota mejorado y actualizado

Kasuk13 committed:
Arreglado valor inicial de fecha de nacimiento en MascotaEdit.js

Kasuk13 committed:
Arreglado error con fechas al registrar y renderizado de servicio en el detalle mascota

Kasuk13 committed:
Agregada Route para validar confirmacion de email

Kasuk13 committed:
Agregado verificación de correo

Kasuk13 committed:
Editado ruta anterior al eliminar pet
